source/extensions/filters/http/grpc_json_transcoder/json_transcoder_filter.cc

```diff
@@ -288,8 +295,14 @@ bool JsonTranscoderConfig::convertGrpcStatus() const { return convert_grpc_statu
 
 ProtobufUtil::Status JsonTranscoderConfig::createTranscoder(
     const Http::RequestHeaderMap& headers, ZeroCopyInputStream& request_input,
-    google::grpc::transcoding::TranscoderInputStream& response_input, TranscoderPtr& transcoder,
-    MethodInfoSharedPtr& method_info) {
+    google::grpc::transcoding::TranscoderInputStream& response_input,
+    std::unique_ptr<Transcoder>& transcoder, MethodInfoSharedPtr& method_info) const {
+
+  if (disabled_) {
```

Comment:
You should guard this before this getting called, in `decodeHeaders`.

Reply:
ping?

Reply:
hey! there is a check for `disabled()` in `decodeHeaders` already (so in the 'regular' flow this will never happen). do you think i should've removed the check from here? did you have different guarding in mind?

Reply:
Yeah if it is checked there, we shouldn't have a check here. What flow hits here?

Reply:
well, this is a public method so theoretically any flow could hit it

Reply:
Generally we don't do defensive coding like this. I would switch this to an ASSERT with a comment if you like.